mp/src/game/shared/momentum/mom_ghostdefs.h

```diff
@@ -297,35 +331,31 @@ struct SavelocReqPacket_t : MomentumPacket_t
     // Stage == 4 ? (The actual saveloc data, in binary)
     CUtlBuffer dataBuf;
 
-    SavelocReqPacket_t(): stage(0), saveloc_count(0)
+    SavelocReqPacket(): stage(0), saveloc_count(0)
     {
-        type = PT_SAVELOC_REQ;
         dataBuf.SetBigEndian(false);
     }
 
-    SavelocReqPacket_t(CUtlBuffer &buf)
+    SavelocReqPacket(CUtlBuffer &buf)
     {
-        type = PT_SAVELOC_REQ;
         stage = buf.GetInt();
-        if (stage == 2)
+        if (stage > 1)
         {
             saveloc_count = buf.GetInt();
-        }
-        else if (stage > 2)
-        {
-            dataBuf.CopyBuffer(buf);
-            // The CopyBuffer method clears the dataBuf and swallows everything inside of buf.
-            // We first need to skip over the 5 bytes used for type + stage
-            dataBuf.SeekGet(CUtlBuffer::SEEK_CURRENT, 5);
-            // And now pull out the saveloc count
-            saveloc_count = dataBuf.GetInt();
-            // And now our buffer's Get is currently in the location for reading the data
+
+            if (stage > 2)
+            {
+                dataBuf.Clear();
+                dataBuf.Put(buf.PeekGet(), buf.GetBytesRemaining());
+            }
```

Comment:
Were these changes tested? Does saveloc requesting work fine?

Reply:
Re: tested this with him on the branch, didn't notice anything out of the ordinary.